lib/interceptor/cache.js

```diff
@@ -84,15 +87,38 @@ class CacheHandler extends DecoratorHandler {
       }
     }
     return this.#handler.onData(chunk)
+    /* 
+      Is 'this.#handler.onData' the previous dispatcher in the chain, e.g. 'redirect'?
+      And in 'redirect.onData(chunk)' it once again calls 'this.#handler.onData(chunk)'. 
+      Would that be 'responseVerify.onData(chunk)'?
+    */
   }
 
-  onComplete(rawTrailers) {
+  onComplete(rawTrailers, opts) {
+    console.log('onComplete, value: ' + this.#value)
+    console.log('onComplete, opts:')
+    console.log(opts)
+
     if (this.#value) {
       this.#value.data.rawTrailers = rawTrailers
       this.#value.size += rawTrailers?.reduce((xs, x) => xs + x.length, 0) ?? 0
+
+      console.log('OnComplete, cache store is being set to: ')
+      console.log([this.#key, this.#value.data, { ttl: this.#value.ttl, size: this.#value.size }])
+
+      /*
+        Why are we setting the cache with the same data as the entry we fetched earlier 
+        from the very same cache?
```

Comment:
This is the new value. What do you mean by fetched earlier? We are here because we didn't have the right entry.